src/edge_detection.cpp

```diff
@@ -350,3 +367,6 @@ void EdgeDetection::getEdgePoints( std::vector < double > & edge, const std::vec
         }
     }
 }
+
+template class EdgeDetection<float>;
```

Comment:
Absolutely!
It is a part of explicit instantiation, which would allow us to separate the template definitions and implementations. Basically, a user of `EdgeDetection` would only need to include `.h` file (which doesn't have implementations) to use the class. 
[Here](https://en.cppreference.com/w/cpp/language/class_template) is more information about them and [here](https://anteru.net/blog/2008/c-tricks-6-explicit-template-instantiation/) is an interesting example on how it can be used.